airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -14,210 +14,282 @@
-        # todo: put this in a password file
         if self.conn.password:
             connection_cmd += ["--password", self.conn.password]
-        if self.lib_jars:
-            connection_cmd += ["-libjars", self.lib_jars]
+        if self.password_file:
+            connection_cmd += ["--password-file", self.password_file]
+        if self.libjars:
+            connection_cmd += ["-libjars", self.libjars]
         if self.files:
             connection_cmd += ["-files", self.files]
         if self.namenode:
             connection_cmd += ["-fs", self.namenode]
         if self.archives:
             connection_cmd += ["-archives", self.archives]
+        if self.num_mappers:
+            connection_cmd += ["--num-mappers", self.num_mappers]
+
+        for key, value in self.properties.items():
+            connection_cmd += ["-D", "{}={}".format(key, value)]
 
-        connection_cmd += ["--connect", "{}:{}/{}".format(self.conn.host, self.conn.port,
-                                                          self.conn.schema)]
-        connection_cmd += cmd
+        connection_cmd += ["--connect", "{}:{}/{}".format(
+            self.conn.host,
+            self.conn.port,
+            self.conn.schema
+        )]
 
         return connection_cmd
 
-    def _import_cmd(self, target_dir,
-                    append=False, type="text",
-                    num_mappers=None, split_by=None):
+    def _get_export_format_argument(self, file_type='text'):
+        if file_type == "avro":
+            return ["--as-avrodatafile"]
+        elif file_type == "sequence":
+            return ["--as-sequencefile"]
+        elif file_type == "parquet":
+            return ["--as-parquetfile"]
+        else:
+            return ["--as-textfile"]
+
+    def _import_cmd(self, target_dir, append, file_type, split_by, direct,
+                    driver):
 
-        cmd = ["--target-dir", target_dir]
+        cmd = self._prepare_command(export=False)
 
-        if not num_mappers:
-            num_mappers = 1
+        cmd += ["--target-dir", target_dir]
+
+        if append:
+            cmd += ["--append"]
 
-        cmd += ["--num-mappers", str(num_mappers)]
+        cmd += self._get_export_format_argument(file_type)
 
         if split_by:
             cmd += ["--split-by", split_by]
 
-        if append:
-            cmd += ["--append"]
+        if direct:
+            cmd += ["--direct"]
 
-        if type == "avro":
-            cmd += ["--as-avrodatafile"]
-        elif type == "sequence":
-            cmd += ["--as-sequencefile"]
-        else:
-            cmd += ["--as-textfile"]
+        if driver:
+            cmd += ["--driver", driver]
 
         return cmd
 
-    def import_table(self, table, target_dir,
-                     append=False, type="text", columns=None,
-                     num_mappers=None, split_by=None, where=None):
+    def import_table(self, table, target_dir, append=False, file_type="text",
```

Comment:
we should add options for extra args people may want to use. one extra args param for export and one for import is what I'd think would make sense

Reply:
Sounds like a good idea